src/daemon/daemon.cpp

```diff
@@ -364,24 +366,54 @@ auto try_mem_size(const std::string& val) -> mp::optional<mp::MemorySize>
 }
 
 std::vector<mp::NetworkInterface> validate_extra_interfaces(const mp::LaunchRequest* request,
+                                                            const mp::VirtualMachineFactory& factory,
                                                             mp::LaunchError& option_errors)
 {
-    // The associated bool indicates whether the interface mode is 'auto', i.e., needs to be configured by cloud-init.
+    std::vector<mp::NetworkInterfaceInfo> factory_networks;
+
+    if (!request->network_options().empty())
+    {
+        try
+        {
+            factory_networks = factory.list_networks();
+        }
+        catch (const mp::NotImplementedOnThisBackendException&)
+        {
+            // If list-networks is not implemented, we should report that --network is not implemented on this backend.
+            throw mp::NotImplementedOnThisBackendException("--network");
+        }
+    }
+
     std::vector<mp::NetworkInterface> interfaces;
 
     for (const auto& net : request->network_options())
     {
+        // Check that the id the user specified is valid.
+        auto pred = [net](const mp::NetworkInterfaceInfo& info) { return info.id == net.id(); };
+        const auto& result = std::find_if(factory_networks.cbegin(), factory_networks.cend(), pred);
+
+        if (result == factory_networks.cend())
+        {
+            mpl::log(mpl::Level::warning, category, fmt::format("Invalid network id \"{}\"", net.id()));
+            option_errors.add_error_codes(mp::LaunchError::INVALID_NETWORK);
+        }
+
+        // In case the user specified a MAC address, check it is valid.
         if (const auto& mac = QString::fromStdString(net.mac_address()).toLower().toStdString();
             mac.empty() || mpu::valid_mac_address(mac))
             interfaces.push_back(
                 mp::NetworkInterface{net.id(), mac, net.mode() != multipass::LaunchRequest_NetworkOptions_Mode_MANUAL});
         else
+        {
+            mpl::log(mpl::Level::warning, category, fmt::format("Repeated MAC address \"{}\"", mac));
```

Comment:
Actually, this is also mistaken.
```suggestion
            mpl::log(mpl::Level::warning, category, fmt::format("Invalid MAC address \"{}\"", mac));
```

Reply:
Right, let me check all comments and warnings. This piece of code suffered lots of changes, so these can be outdated.

Reply:
Done. I understand this PR is not about MAC checking, but the comment and warning we corrected were in the diff, so I consider OK to fix it here.

Reply:
Yes, that makes sense. This PR involved changing the severity in a network check and the mac check can be regarded as a necessary piece for consistency. However, it is an untested piece... Care to add a test akin to `refuses_launch_with_invalid_network_interface` but checking mac validity (so `refuses_launch_with_invalid_mac_address` ? Thanks!

Reply:
Hey Luis, I chanced upon daemon tests and realized that the missing test here would not be easy after all. I thought it would be the same as the tests you already added, only with a bad MAC replacing the bad network ID. But that's not the case.

It is true that the repeated MAC check derives from having 2 interfaces at different levels (CLI and RPC). Although both should normally be tested, we have currently no way to test the daemon directly via RPC without going through a (test) client. But that test client reuses most of the real client, and that includes the mac check. 

So we have no way to send a bad MAC to the daemon, and I hadn't realized that. It could be done, but it would require re-implementing a launch cmd or otherwise hooking to the RPC. That is an open problem and not something that should be done for this. So sorry about the trouble.